### PR TITLE
py-azure-...: add new versions

### PR DESCRIPTION
**Pull request by w8jcik**
Needed by `azure` storage plugin of Snakemake 8.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py [hunk 1]
```diff
@@ -40,7 +48,7 @@ class PyAzureStorageBlob(PythonPackage):
     depends_on("py-typing-extensions@4.3:", type=("build", "run"), when="@12.17:")
     depends_on("py-typing-extensions@4.0.1:", type=("build", "run"), when="@12.15:")
     depends_on("py-isodate@0.6.1:", type=("build", "run"), when="@12.15:")
+    depends_on("py-msrest@0.7.1:", type=("build", "run"), when="@12.14")
+    depends_on("py-msrest@0.6.21:", type=("build", "run"), when="@:12.14")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: Both of these include 12.14, might be a mistake here (I realize the existing code already contained this mistake)

**Review by adamjstewart — COMMENTED** [review]
Style comment: I usually don't include the URL of the file that contains deps, I only include the filename, and only if it's different than the usual suspects. Also, I usually put build deps like setuptools before run deps. And I usually put `url_for_version` after all other vars/directives and before all functions. But these are all optional style things, just want to make sure you aren't working extra hard to update things you don't need to.

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py (hunk 1 above)
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: Fixed

**Comment by w8jcik** [comment]
> Style comment: I usually don't include the URL of the file that contains deps

Let's take a look at `py-azure-storage-python`.

The `homepage` is https://github.com/Azure/azure-storage-python. But the actual package and `setup.py` is located at 
https://github.com/Azure/azure-sdk-for-python/blob/azure-storage-blob_12.19.0/sdk/storage/azure-storage-blob/setup.py

It is difficult to find it without the comment. Such URL also hints what are the tags in this repository, eg. `azure-storage-blob_12.19.0`.

It is not trivial to figure it out without the comment.

**Comment by w8jcik** [comment]
> Also, I usually put build deps like setuptools before run deps.

Corrected

**Comment by w8jcik** [comment]
> And I usually put url_for_version after all other vars/directives and before all functions.

Reordered

**Comment by adamjstewart** [comment]
> Let's take a look at py-azure-storage-python.

I've never looked at the GitHub before. I always download the release from PyPI, where the pyproject.toml/setup.py/setup.cfg is guaranteed to be in the root directory.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**